Sources/StreamChat/StateLayer/ChatClient+Factory.swift

```diff
@@ -34,9 +34,8 @@ extension ChatClient {
     ///
     /// - Throws: An error while communicating with the Stream API.
     /// - Returns: An instance of ``ChannelList`` which represents actions and the current state of the list.
-    public func makeChannelList(with query: ChannelListQuery, dynamicFilter: ((ChatChannel) -> Bool)? = nil) async throws -> ChannelList {
-        let channels = try await channelListUpdater.update(channelListQuery: query)
-        let channelList = ChannelList(initialChannels: channels, query: query, dynamicFilter: dynamicFilter, channelListUpdater: channelListUpdater, client: self)
+    public func makeChannelList(with query: ChannelListQuery, dynamicFilter: ((ChatChannel) -> Bool)? = nil) -> ChannelList {
+        let channelList = ChannelList(query: query, dynamicFilter: dynamicFilter, channelListUpdater: channelListUpdater, client: self)
         syncRepository.trackChannelListQuery { [weak channelList] in channelList?.state.query }
         return channelList
     }
```

Comment:
- Drops `async` and `throws` 
- Server sync is not forced, one needs to use `loadChannels()` function for fetching server state

Reply:
We need to think this one through. Maybe we keep both options? I can image customers forgetting to call `loadChannels` and complaining about why they don't get the latest data.

Reply:
Could happen. I'll think about naming here.

Reply:
Not sure if there is better naming, open for ideas.
```
let a = makeChannelList(with: ChannelListQuery(filter: .noTeam))
let b = try await makeChannelList(refreshing: ChannelListQuery(filter: .noTeam))
```

Reply:
The `make` functions should never make any calls. If they do, it will be misleading. Factory methods should not have any side effects. Just create a function.

Usually, you don't want it to perform any calls initially. You want to create it, put it in a view, and only when the view is loaded do you perform a call. 

If for some reason, we want to provide a way to create a channel list, and at the same time initialise the loading, we should name it something else then `make`. Although I don't see the need having this TBH.

Reply:
For me, this is what makes more sense:

> Drops async and throws
Server sync is not forced, one needs to use loadChannels() function for fetching server state

Reply:
Sounds like I should take a step back, look at the ChatClient+Factory as a whole, and come back with a proposal for all the methods we have there.

Channel related methods do require sync with a server in some cases so I need to think about it. Currently these are also named as `makeXYZ`.

Something along the lines:
- `makeXYZ` creates the object, does not do any refresh calls
- `somethingSomething` syncs with server and then returns the object

I'll look around in Apple APIs etc for inspiration for the second case.

Reply:
Seems like there are 3 cases:
1. Give me an instance which represents the local state (sync)
2. Give me an instance which represents the server state (async, throws)
3. Create something new on the server if needed (async, throws)

Reply:
![Uploading Screenshot 2024-04-12 at 10.54.33.png…]()

Reply:
For me `load` makes most sense. In any case, we should also expose a `watch` method and be consistent with the other SDKs: https://getstream.io/chat/docs/android/watch_channel/?language=kotlin&q=watch

Reply:
`Chat.watch()` is there. 
Naming is hard since some of the methods actually create a new thing on the backend side (direct channel/chat case).

Reply:
Actually, we can use `get` and `getOrCreate`. We use both on video, and also the API is using this convention.

`chat.get` and `chat.getOrCreate` - I think this is the best one.

Reply:
Concrete examples:

**make+retrieve**
```swift
// Give me channel list with the local state
func makeChannelList(…) -> ChannelList

// Give me channel list with the server state
func retrieveChannelList(…) async throws -> ChannelList

// Create a new direct message channel or if it exists, give me the one we already have
func retrieveDirectMessageChat(…) async throws -> Chat
```

**make+fetch**
```swift
// Give me channel list with the local state
func makeChannelList(…) -> ChannelList

// Give me channel list with the server state
func fetchChannelList(…) async throws -> ChannelList

// Create a new direct message channel or if it exists, give me the one we already have
func fetchDirectMessageChat(…) async throws -> Chat
```

**get+getOrCreate**
```swift
// Give me channel list with the local state
func getChannelList(…) -> ChannelList

// Give me channel list with the server state
func getOrCreateChannelList(…) async throws -> ChannelList

// Create a new direct message channel or if it exists, give me the one we already have
func getOrCreateDirectMessageChat(…) async throws -> Chat
```

Reply:
no, it should be sth like this:
```
let chat = client.makeChat(...) // sync
chat.get()
chat.getOrCreate()
```
Check the call docs in video: https://getstream.io/video/docs/ios/guides/joining-creating-calls/

Reply:
This would be going back to `synchronize()` and having channel id check in every method call.

Reply:
other SDKs seem to use `queryChannelList()` and `queryChannel()` right? There seems to exist a lot of different patterns in the SDK and we should start to align this now. 

For Chat:
- makeChat()
- query() / fetch() / get()
- watch()

For ChannelList:
- makeChannelList()
- query() / fetch() / get()
- watch()
- queryOrCreate() / fetchOrCreate() / getOrCreate()

But since @martinmitrevski is saying that `get` is what it is used on video, then the get is not the preferrable way?